en/docs/setup/migrating-preparing-for-migration.md

```diff
@@ -21,6 +21,113 @@
 
 With WSO2 Identity Server 5.11.0, groups and roles are separated. For more information, see [What Has Changed in 5.11.0](../../setup/migrating-what-has-changed#group-and-role-separation).
 
+## Groups and Roles Improvements Migration
+
+With WSO2 Identity Server 5.12.0, groups and roles improvements are introduced. For more information, see [What Has Changed in 5.11.0](../../setup/migrating-what-has-changed#roup and role separation Improvements).
+
+Please note that the abbreviations mentioned in that section is also used in here.
+
+Please refer to the below sections related to this improvement.
+
+### Claim configuration changes
+
+#### Following changes are added to the product claims.
+
+* Wso2.role claim - Modified
+  * Remove `supported by default`
+  * Change display name to `Roles and groups`
+  * Change description to `Include both userstore groups and internal roles`
+* Wso2.roles claim - New
+  * Display name : Roles 
+  * AttributeID:  roles
+  * Description: Roles
+  * Add `Supported by default`
+  * Add `read-only`
+* Wso2.groups claim - Modified
+  * Add  `supported by default`
+  * Add  `read-only`
+* OIDC group claim - Modified
+  * Change mapped local claim to wso2.groups claim
+* OIDC roles claim - New
+  * Mapped to the local claim, wso2.roles
+* SCIM2 roles.default claim - Modified
+  * Change mapped local claim to wso2.roles claim
+    
+#### Migration preparation for claim changes
+* All claim configurations are already configured OOTB in the fresh pack, and will be done via the 
+  migration client for migrating deployments. No need to configure these manually.
+* Any custom external claim mapped to the wso2.role claim should be mapped to either wso2.roles 
+  or wso2.groups claim as per the requirement of the custom use case.
+* Our recommendation is to fix any consuming client to become compatible with these changes. 
+  But if somehow the above configs need to be reverted(possibly in a migrated deployment), it can 
+  be done via calling IS server REST APIs(https://is.docs.wso2.com/en/latest/develop/claim-management-rest-api/).
+  
+### Utilizing carbon kernel level support
+
+The following abstract userstore manager APIs: getUserClaimValues(), getUsersClaimValues(), 
+getUserClaimValuesWithID(), getUsersClaimValuesWithID() now support both wso2.roles 
+and wso2.groups claims properly. We recommend modifying custom extensions to request wso2.roles or wso2.groups
+via above APIs rather depending on wso2.role claim.
+
+### Service provider role mapping, and IdP role mapping  are restricted strictly to role mapping
```

Comment:
this doesn't seem to be a topic

Reply:
```suggestion
#### Service provider role mapping and identity provider role mapping restrictions
```

Reply:
Here we mention each new change introduced, and this needs to be a topic in that sense

Reply:
Fixed with fa6da06b8ecc8fd6ac34122f47ef8e9050296214